core/runtime.rs

```diff
@@ -689,14 +692,20 @@ impl JsRuntime {
 
     // Check if more async ops have been dispatched
     // during this turn of event loop.
-    if state.have_unpolled_ops {
+    // If there are any pending background tasks, we also wake the runtime to
+    // make sure we don't miss them.
+    // TODO(andreubotella) The event loop will spin as long as there are pending
+    // background tasks. We should look into having V8 notify us when a
+    // background task is done.
+    if state.have_unpolled_ops || has_pending_background_tasks {
```

Comment:
I'm aware of this limitation, but I haven't found any method in V8 API that would do that

Reply:
I looked up implementation of `HasPendingBackgroundTasks()` and it is:
```c++
bool Isolate::HasPendingBackgroundTasks() {
#if V8_ENABLE_WEBASSEMBLY
  i::Isolate* isolate = reinterpret_cast<i::Isolate*>(this);
  return isolate->wasm_engine()->HasRunningCompileJob(isolate);
#else
  return false;
#endif  // V8_ENABLE_WEBASSEMBLY
}
```
Ie. only WASM compilation is considered as background pending task, I think we can leave this bit of code as is